tests/bsim/bluetooth/host/gatt/ccc_value/testlib/adv.c

```diff
@@ -0,0 +1,83 @@
+/* Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/bluetooth/l2cap.h>
+#include <zephyr/bluetooth/uuid.h>
+#include <zephyr/kernel.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/__assert.h>
+
+struct bt_testlib_adv_ctx {
+	struct bt_conn **result;
+	struct k_condvar done;
+};
+
+/* Context pool (with capacity of one). */
+static K_SEM_DEFINE(g_ctx_free, 1, 1);
+static K_MUTEX_DEFINE(g_ctx_lock);
+static struct bt_testlib_adv_ctx *g_ctx;
+
+static void connected_cb(struct bt_le_ext_adv *adv, struct bt_le_ext_adv_connected_info *info)
+{
+	k_mutex_lock(&g_ctx_lock, K_FOREVER);
+
+	if (g_ctx->result) {
+		*g_ctx->result = bt_conn_ref(info->conn);
+	}
+	k_condvar_signal(&g_ctx->done);
+
+	k_mutex_unlock(&g_ctx_lock);
+}
+
+int bt_testlib_adv_conn(struct bt_conn **conn, int id, uint32_t adv_options)
```

Comment:
```suggestion
int bt_testlib_adv_conn(struct bt_conn **conn, uint8_t id, uint32_t adv_options)
```
Suggest to use same type here as the `bt_le_adv_param.id`